webapp/src/ts/services/z-score.service.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@angular/core';
+
+import { ChangesService } from '@mm-services/changes.service';
+import { DbService } from '@mm-services/db.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ZScoreService {
+  private tables;
+  private readonly CONFIGURATION_DOC_ID = 'zscore-charts';
+  private readonly MINIMUM_Z_SCORE = -4;
+  private readonly MAXIMUM_Z_SCORE = 4;
+
+  constructor(
+    private changesService:ChangesService,
+    private dbService:DbService,
+  ) {
+    this.changesService.subscribe({
+      key: 'zscore-service',
+      filter: (change) => change.id === this.CONFIGURATION_DOC_ID,
+      callback: () => this.init(),
+    });
+  }
+
+  private findTable(id) {
+    for (let i = 0; i < this.tables.length; i++) {
```

Comment:
Maybe we can change this one for Array's forEach()?

Reply:
I think the idea behind using a for is to return early, as soon as we found the first match.

Reply:
This PR is already merged, so no need to go update at this point, but how about the following?

```
this.tables.find( (t) => t.id === id);
```

What I like most about the terse syntax is that this is easy to scan and get an idea of what's going on ("find the item that matches the id") without needing to read/scan/parse the looping business.